src/consensus/aft/test/main.cpp

```diff
@@ -7,8 +7,8 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
 #include <doctest/doctest.h>
 
-threading::ThreadMessaging threading::ThreadMessaging::thread_messaging;
-std::atomic<uint16_t> threading::ThreadMessaging::thread_count = 1;
+std::unique_ptr<threading::ThreadMessaging>
```

Comment:
I suspect there's a good reason why this isn't possible, but instead of having to declare and define one of these for every object file, could there be a single static instance of it in `thread_messaging.h` so that including `thread_messaging.h` is enough to be able to schedule async messages?

Reply:
No - we can (and do!) declare but not define the static instance in the header. The neater solution would be to define it once in a separate module, and link that to these tests, but we've stuck to larger single binaries rather than standalone linkables for the most part.